### ISSUE TITLE
Auto agent update issue, 32bit win agent

### ISSUE THREAD

**Issue by Allowee** [Ylianst/MeshCentral#3016]
Updated to the lastest version yesterday, 0.8.99
Today I noticed that the 32bit windows mesh agent clients are no longer working.
They are online, but no desktop, terminal or files tab.
Tried to send commands in the Console, no repsonse.

After using a different method for logging in I find that the update file hangs.
meshagent.update.exe is 1,6mb and there is a meshagent.update.exe_unzipped at 0kb.
After a service stop the .exe_unzipped file becomes 3,4mb and when I rename it and run it will update the agent just fine.

![image](https://user-images.githubusercontent.com/55733884/128839818-a893b007-539b-4f94-ae2b-925b61b0a920.png)

Also in the log file
[2021-08-10 10:13:19 AM] [8F9F521A46814C2D] ..\microstack\ILibParsers.c:2711 (0,0) FATAL EXCEPTION @ [FuncAddr: 0x013bdf28 / BaseAddr: 0x01411873 / Delta: 342347]

**Comment by Allowee** [comment]
Just saw other reports about this, but none mention the .exe_unzipped and non updating agent.

**Comment by rodrigolzd** [comment]
Allowee, you are right about this. I saw this as well, but only after I posted the issues.

**Comment by Ylianst** [comment]
I tested the update path for 32bit agent a bunch, I have not seen this. Bryan will look at it. The error message in the log:

```
meshservice -fdelta 342347
[BaseAddr: 0x00f51873, duk_hobject_find_entry => c:\meshagent\meshagent\meshservice\duk_hobject_props.c:1389]
```

**Comment by Ylianst** [comment]
Bryan has a theory that this problem will only occur on Windows 7 and maybe only Windows 7 SP1 for both 32bit and 64 bit agents. This is due to an OS bug in Win7 SP1. We need confirmation that this theory is correct.

If correct, we can't reproduce this issue and Bryan is working on a fix for this. A server update will fix this.

**Comment by Allowee** [comment]
the server version was fairly recent,, fresh install 11 days ago.
OS versions differ, I was thinking about  permissions but 1 is even running as administrator.

![image](https://user-images.githubusercontent.com/55733884/128907157-4a61706c-d8f2-4404-a90f-3bdd8a93f169.png)

**Comment by krayon007** [comment]
I'll run some tests with the 32 bit agent on those platforms... A few days ago, I found a bug in the Win32 API that caused self-update to fail in the exact way you described, but I was only able to reproduce it on Windows 7. I'll do some more testing to see if this same issue appears on those other platforms.

**Comment by Ylianst** [comment]
Thanks. So the problem occurred with all 3 devices that had the 32bit Agent?

As a temporary workaround, you can click on a device, go to the "Console" tab. Hit "Agent Action" and select "Upload Recovery Core"

![image](https://user-images.githubusercontent.com/1319013/128908945-d24b2b85-8480-451b-b409-e30a14c3c200.png)

When type "AgentUpdate".

![image](https://user-images.githubusercontent.com/1319013/128909217-91e98d34-4746-4bf1-9aae-80dd1919b1b4.png)

If you type "versions" before and after "agentupdate", you should see the agent change to the new version and everything work.

**Comment by Allowee** [comment]
Yes I have 3 32but agents (2 are on 32bit os) 
I was unable to get a response on the console on all three, however the upload default core did work for 1 client I tested it with.
I did try to restart the service and it downloaded the update, yet it did not install it. (Is the 'agentupdate' command different from the auto update after agent restart?)

**Comment by krayon007** [comment]
For the agent where you were able to load a default core, was that the 64 bit windows running a 32 bit agent?
The 'agentupdate' command is different from the normal self-update, as it's a completely different code path. For the one that you can load the default core, can you load the recovery core instead, and then run 'agentupdate' on the console tab afterwards?

are you able to load the recovery core for any of the agents that didn't work?

**Comment by Ylianst** [comment]
If you can provide as exact information as possible, that would be great. We need to be able to reproduce this on our systems. Any exact steps we can take to make this happen on our development machines would solve most of the issue. So far, we can't reproduce this.

Yes, the "agentupdate" command is a different update path for the agent then the full automatic path. We have two different ways of updating the agent.

Lastly, in this picture:

![image](https://user-images.githubusercontent.com/1319013/128915630-5220ec6c-4791-43ff-9240-d7545d17926f.png)

All agent versions show "Aug 5 2021", so all agents are updated to the latest version. Are you still getting the missing tabs on these agents? You can type "versions" once a meshcore is loaded to see the agent version.

![image](https://user-images.githubusercontent.com/1319013/128915826-d0311277-6cac-4867-b514-c19629cfe90e.png)

**Comment by Allowee** [comment]
The one I tried to load the default core was the top one, win 10 Enterprice.
I also just tried the recovery core on all three now, all three worked ok (desktop tab went away) did the 'agentupdate' on all three and all restarted fine. (timestamp on meshagent.exe updated)

At this moment the three system run ok and update ok. all other systems updated just fine.
I expect the problem to only have been in the old agent, the agent was a version from 8 april 2021.

I needed the systems so I updated them manually, would have been smart to keep a copy of that meshagent version..

**Comment by Ylianst** [comment]
Bryan has been trying various operating systems and can't find any issues. Since we don't have sufficient information to replicate this, I am going to close this issue. I am adding a new option in MeshCentral v0.9.1:

```
{
  "settings": {
    "agentUpdateSystem": 2,
  }
}
```

This new setting will force the server to always use the "meshcore" update system for updating agents.

**Comment by Allowee** [comment]
I have found the 32bit installer I used for this system.
uninstalled the meshagent from windows.
Ran the old installer and installed.
The meshagent updated itself to the latest version without a problem, all tabs are there no issue at all.

I'm totally confused as I can't replicate it with the original meshagent installer now.
Really strange this.

I will try to boot another old system with Windows Embedded Standard [7601] and see how that update goes
Sorry to have wasted your time
